package.json

```diff
@@ -38,7 +38,7 @@
     "lint-markdown-fix": "yarn lint-markdown --fix",
     "lint-markdown": "markdownlint \"**/*.md\" --config=.markdownlint.json --ignore-path=.markdownlintignore",
     "lint": "cross-env NODE_OPTIONS=\"--max-old-space-size=16384\" eslint .",
-    "postinstall": "yarn patch-package && yarn husky install && yarn build",
+    "postinstall": "tools/scripts/postinstall.sh",
```

Comment:
Using a shell script isn't great because it isn't portable to Windows.

Is there another way we can wire this up?

Reply:
Ah, damn. Open to any suggestions, I've just found that having the ability to ensure postinstall steps aren't run when automated installs happen as part of migrations helps make sure they always works. So this simple env var + shell script approach was what I settled on at the time

Reply:
ts-node + child process is one option?

Reply:
I don't know a good solution - maybe just a ts-node script is the best option? ugh. would need to do some googling.

i know we do have contributors that work on windows so we'd need to solve this.

perhaps we can just do both a `.bat` and `.sh` version of the script and call the relevant one from a simple ts-node script? IDK

Reply:
Could you write the script as a Node `.js` script? If so, you could leverage it in `postinstall` much like the `decorate-angular-cli.js` file works for an Nx Angular repo:

```json
{
  ...,
  "scripts": {
    ...
    "postinstall": "node ./decorate-angular-cli.js && ngcc --properties es2020 browser module main",
    ...
  },
  ...
}
```

Reply:
https://github.com/google/zx looks pretty sweet 😎

Reply:
Holy shit zx is amazing.
I wish I'd learned about it sooner!!!

Reply:
I did have a play with zx but it's ESM only and therefore requires some additional fiddling with tsconfigs and package.jsons

We already have `execa` in our dependency tree so I decided to formalize that devDependency instead and made a small wrapper `$` function which matches `zx` usage and output